### ISSUE TITLE
[Suggestion] Barrels for Beam weapons too

### ISSUE THREAD

**Issue by Kaiser-Indrasil** [tsunamayo/Starship-EVO#2677]
Aside from the whole weapon balance discussion, I think it would be cool to have the option to equip the beam weapons with their own barrels. Right now they all have uniform Range and Fire Time. In order to keep their balance right, you'd have to introduce the Spread for beam lasers too. Beams could use same models of barrels as Lasers do, but they would grant different stats to Beams:
- Gatling Barrel - fires a burst of 4 beams in a short succession. Shorter Range, bigger Spread
- "Recoil" (more like, Long Charge) Barrel - longer charge and fire times
- Cooling Barrel - self-explanatory
- Range Barrel - longer Charge Time and Range, but much shorter Fire Time to prevent making it OP.

**Comment by ZachZent** [comment]
I believe the plan is that laser bolts will have modules and barrels, laser beams will have just modules, and kinetics will have just barrels. That being said I would love to see those options for beams as well.

___

And just an aside, maybe one way to fix a potential hitscan problem before it comes up is the beams being less like laser pointers and more like spartan lasers.

https://youtu.be/dn3-M6tsFjo

Hold down the fire button and after a short time a powerful burst will fire before taking an amount of time to cool down. This completely removes any sort of problem with hitscan rather than needing to balance turret turn rates or damage rates to prevent massive, perfectly tracking hitscan turrets that can one-shot fighters. The larger the weapon, the longer the charge up and/or cool down time making it a more anti-cap ship weapon. 

The modules wouldn't have to change and can still do what they do. As for keeping with the post with barrels...
Gatling: Rather than one burst, it will fire multiple busts at once like a shotgun. Still need to charge/cool down
Recoil: With the spartan laser in halo, after ever shot there is a large amount of upward recoil requiring the player to reset their aim. Perhaps the laser can have the same effect with the recoil dampening that a bit
Cooling: Either cooldown rate is reduced or charge time is reduced
Range: Obvious

**Comment by Kaiser-Indrasil** [comment]
Yeah, I would definitely like a spool-up time introduced! It could display a very thin line of where it's going to fire, in anything from 0.5 seconds to 5 seconds, in order to give the small and nimble fighters - friend and foe - a chance to escape the deadly beam's path. Otherwise this would happen:
https://www.youtube.com/watch?v=khIWdolT9xY&feature=youtu.be&t=88
Poor bomber pilots...

Here is a good example of how the beam fire warning could be implemented. The stronger the beam, the longer the spool-up and warning time would be:
https://www.youtube.com/watch?v=wOy1RQ1DASo

And here is a different clip of what I mean by "Gatling Beams". They should blink with a lower frequency in order to keep them in balance with the Laser Gatling though:
https://www.youtube.com/watch?v=6sWdognTGAg
Left and right ones use Pulse Lasers ("Gatling" Beams in Starship EVO), middle one uses Beam Lasers (normal Beams in Stevo).

**Comment by ZachZent** [comment]
I would say no to a guiding laser. Firstly because if players use a lot of beams, it can become a laser show. Second other ships wouldn't really see it minus a few particular instances. As for that video, you are in a small fighter flying in a straight line right at the enemy ship which has a massive fuckoff laser? Yea, I'm blaming the pilot on that one :D

I can see what you mean by gatling beams in that video, I just wonder what would be the gameplay difference between three bolts and one beam? More over, I think that also would go into a bit too far into the laser bolt category.

**Comment by Kaiser-Indrasil** [comment]
Damn, you've made me dig quite deep to provide a source material ;) Here:
https://youtu.be/oCfYLGBkzZI?t=7047
You can see that a single beam salvo has a chance to hit a small and nimble target multiple times, provided that the turret rotates fast enough.

**Comment by ZachZent** [comment]
Ah ok, I see what you mean now
